### ISSUE TITLE
Please make it so I do not have to craft each item for a crafting quest

### ISSUE THREAD

**Issue by freshwawa** [GTNewHorizons/NewHorizons#5088]
#### Which modpack version are you using?
Version: 2.0.7.5-1.7.10
#
#### If in multiplayer; On which server does this happen? 
Prospercraft
#
#### What did you try to do, and what did you expect to happen?
In many of the bqq quests, when you are required to make an item, you need to craft each piece individually for the questbook to recognize that you have the component. This is really time consuming and makes the pack a lot less enjoyable for me because I usually craft in bulk just so that I can avoid having to make something over and over again, only to have the book not let me proceed with quests until I craft an iron plate or iron rod for the millionth time in a row. 
#
#### What happened instead? (Attach screenshots if needed)

#
#### What do you suggest instead/what changes do you propose?
Please have the quest book recognize items in your inventory instead of only after you have crafted them.

**Comment by Dream-Master** [comment]
We have this discussion allready. Low tier Quests are crafting Quests and I won't change it. When you have machines and automation in LV Age Quests not longer crafting Quests.

**Comment by bigbass1997** [comment]
Dream, What is your reason for these to be crafting tasks? Like what does it achieve forcing them to be that way?

Couldn't find any other issues that expressed the reasoning; if there is, just reference it so people know what "discussion" you're talking about.

**Comment by 0lafe** [comment]
It is kind of the only thing standing in the way of cheesing the party system for quest rewards. If everything is detect only, you only need to make 1 thing to get as many rewards as you have people.

**Comment by bigbass1997** [comment]
But according to that logic, you should make every quest in the game be crafting tasks. If people want to cheat the party system that's their own issue. Though are there really any rewards in early game that would be horrible if multiple people in the same group got the rewards?

**Comment by draknyte1** [comment]
> if multiple people in the same group got the rewards?

We already removed party loot.. so it’s an invalid argument on that front. Beyond LV if people aren’t partied but sharing their stuff for quest completion, why not let people do it during their start quests as well? It’s tedious and everyone gets ahead of themselves and has to remake more fire bricks or coke bricks. (Most common two I see people complain about)

**Comment by 0lafe** [comment]
They probably would be if they didn't suck to complete. Having the majority of early game quest be crafting ones means that you need to actually do them per person if you want to cheat the rest of the quests. 

>Remove loot

Yea uh, that's the point? With non crafting quests, people in separate parties, playing together, function the same as people in a party that shares rewards.

**Comment by bigbass1997** [comment]
So the only reason the beginning quests are crafting tasks, is because there's some kind of worry that people in a group will get multiple rewards? Again I ask, are the rewards so rare that this is actually a problem?

The OP is not alone, this is a very common complaint and I've suffered it myself too. Crafting tasks are annoying, don't even work all the time, and to me it seems they serve no real benefit either.

**Comment by draknyte1** [comment]
> The OP is not alone

The amount of people who I just suggest edit their quest file and just replace all the crafting is silly. Beyond Dream wanting it this way ‘just because’, I’ve never heard any real logic behind the argument for it.

**Comment by Methes** [comment]
I vaguely remember the logic being some like "they will learn how to craft those things if it's a crafting task" which doesn't sound very valid. I'm probably wrong though.

**Comment by Dream-Master** [comment]
Yes true at the beginning it will teach every player how to craft some stuff.
And the party rewards are not disabled in lastest BQ3 build. (was disabled in BQ 1)

**Comment by bigbass1997** [comment]
Even so, that doesn't warrant them to be crafting tasks; they could be retrieval and still show the player the steps needed. (even though NEI already does that for you) In many cases, if not most, there's only 1 way to craft the end result, and you'd easily be able to look that up via NEI.

**Comment by observeroftime02** [comment]
I see another problem with some of these retrieval tasks even. Let's look at the quest "another brick in the wall" (number 25) to illustrate the point I'm about to make. 

The quest has you retrieve 104 unfired coke oven bricks, then has you retrieve 104 coke oven bricks. The issue with quests of this type is that you can not progress on crafting unless you first make all 104 unfired coke oven bricks. That goes against the way a lot of people play, namely craft a batch of unfired bricks, put them in the oven, then craft the next batch so that the bricks are firing as they are being crafted by the player. If you get ahead of yourself and start processing the unfired bricks while you're crafting them, you can not complete the quest the proper way, and have to craft extra bricks just for completions sake. 

IMHO, having this be a retrieval task for 104 coke oven bricks would be fine, since having 104 coke oven bricks implies that at some point, you have had the 104 unfired coke oven bricks. I'm sure there are more quest of this variety that have you first craft all the intermediary components before crafting the end result that could be looked at. I don't see a compelling reason why the quest is the way it is, and I certainly see why a lot of people get hung up on this particular quest. It creates more frustration with the questing mechanics than it actually helps anyone I think.

**Comment by bigbass1997** [comment]
observeroftime, yes that's another issue too, but different than the one at hand here. I completely agree with you though, and there's quests like that at higher tiers too.

**Comment by freshwawa** [comment]
> Yes true at the beginning it will teach every player how to craft some stuff.
> And the party rewards are not disabled in lastest BQ3 build. (was disabled in BQ 1)

No offense but that is not really on you to teach us. We will have to craft them whether you make the quest that way or not. It's not like we can't simply look it up on NEI each time... I see no reason for it to be this way other than the fact that you like it that way, and that's honestly not the best way to take to things as a dev. From the responses here alone it seems I am not the only one with this complaint by a longshot. 

Can you provide a real reason as to why you want to keep it this way? For instance, I had 4 furnaces crafted because I was using them early in the game. Then I had recipes later needing furnaces and I wanted to use those furnaces I had initially made... but I couldn't, so I had to go and make one each time. Do you not see how frustrating that is? I can't even use the furnaces I have to upgrade into other items. I literally just have to trash them or hope that I will need to make more than one. 

Same with iron rods, screws, and plates. What is the point of requiring us to craft these over and over again for each quest? Does it not seem odd to you that literally no one plays Minecraft this way, crafting only what they need each time they are going to craft something? It is nothing but a hindrance to the way people want to enjoy the modpack. If you insist on leaving it as an option, at least give people the option to turn it off.

**Comment by grompe** [comment]
If Better Questing mod counted crafted items even for the locked quests, this wouldn't be an issue.
HQM did that IIRC.

**Comment by Anarack** [comment]
My issue isn't so much the crafting in general, its mainly that some quest should be detect due to being able to find them inworld without a massive amount of searching.

**Comment by richardhendricks** [comment]
We changed a lot of the findable-in-world/from-loot-bag items to retrieval. If there are any other *specific* items that should be retrieval based on that, file a separate issue.
